### PR TITLE
Fix message type inconsistency between locales

### LINKED ISSUE #119557: [gen-l10n] wrong generation when the placeholder is missing in other languages
Post 3.7 upgrade, getting the following error for all the i10n strings.

```
Error: Can't declare a member that conflicts with an inherited one.
.dart_tool/…/gen_l10n/app_localizations_ur.dart:17
  String get devOptionsEnabled => '';
             ^^^^^^^^^^^^^^^^^
: Context: This is the inherited member.
.dart_tool/…/gen_l10n/app_localizations.dart:140
  String devOptionsEnabled(String appName);
         ^^^^^^^^^^^^^^^^^
: Error: Can't declare a member that conflicts with an inherited one.
.dart_tool/…/gen_l10n/app_localizations_bn.dart:449
  String get pulseChlngAcc => '';

             ^^^^^^^^^^^^^
: Context: This is the inherited member.
.dart_tool/…/gen_l10n/app_localizations.dart:1004
  String pulseChlngAcc(String player);

         ^^^^^^^^^^^^^
Target kernel_snapshot failed: Exception


FAILURE: Build failed with an exception.

* Where:
Script 'C:\Users\Shikhar\Documents\Github\flutter\packages\flutter_tools\gradle\flutter.gradle' line: 1151
```

The code gen takes place after running the command _flutter gen-l10n_ as [per this guide](https://docs.flutter.dev/development/accessibility-and-localization/internationalization).

Things I've tried:
1. Uninstall flutter, reinstall 3.7 fresh
2. Flutter clean

No code changes in project since the upgrade.

### PR DESCRIPTION
**Pull request by thkim1011**
If a message does not use placeholders in some language, it will be defaulted to a getter even if it makes use of the placeholder in other languages. This ultimately breaks the code because locales extend a common base class.

I also had to fix some issue with error handling that may have been introduced with https://github.com/flutter/flutter/pull/119644 as code generation errors and syntax errors are handled slightly different, and I forgot to handle the latter case with the previous PR.

Fixes #119557.

### CODE REVIEW

**Comment by AlexV525** [comment]
Tests are complaining about differences in generated files, looks like other languages are using the English version. 
image

**Comment by christopherfujino** [comment]
@AlexV525 I would appreciate your review on this one

File: packages/flutter_tools/lib/src/localizations/gen_l10n.dart [hunk 1]
```diff
@@ -877,6 +877,7 @@ class LocalizationsGenerator {
     _allMessages = _templateBundle.resourceIds.map((String id) => Message(
        _templateBundle, _allBundles, id, areResourceAttributesRequired, useEscaping: useEscaping, logger: logger,
     )).toList();
+    hadErrors = _allMessages.any((Message message) => message.hadErrors);
```
Comment (AlexV525) [inline packages/flutter_tools/lib/src/localizations/gen_l10n.dart:880]: Can we address valid tests for handling errors?

**Review by AlexV525 — APPROVED** [review]
LGTM with messages changes

**Review by christopherfujino — APPROVED** [review]
LGTM

**Comment by pedromassango** [comment]
@christopherfujino could we promote this for hotfix on stable as it is a breaking change? 👀

**Comment by AlexV525** [comment]
> @christopherfujino could we promote this for hotfix on stable as it is a breaking change? 👀

Submitted #120679.

**Pull request merged**